sdk/tests/conformance2/textures/misc/copy-texture-image-luma-format.html

```diff
@@ -0,0 +1,245 @@
+            format: gl.LUMINANCE_ALPHA,
+            width: 64,
+            height: 32,
+            depth: 2
+        }
+    ];
+
+    testGroup.forEach(function(testcase) {
+        debug("");
+        debug("Testing copytexsubimage3d_luma_format_" + testcase.name);
+
+        var texture = [];
+        texture[0] = gl.createTexture();
+        texture[1] = gl.createTexture();
+        texture[2] = gl.createTexture();
+        var bytes;
+        var layer = 0;
+        var width = testcase.width;
+        var height = testcase.height;
+        var depth = testcase.depth;
+        var w = testcase.width;
+        var h = testcase.height;
+        var format;
+        var msg;
+        var uint = new Uint8Array(width * height * 4);
+        var j = 0;
+        for (var i = 0; i < uint.length - 1; ++i) {
+            uint[i + 1] = (uint[i] + 10) % 255;
+        }
+        var uint2 = new Uint8Array(width * height * depth * 4);
+        for (var i = 0; i < uint2.length; ++i) {
+            uint2[i] = 0xFF;
+        }
+
+        for (var i = 0; i < 2; ++i) {
+            var program;
+            if (i == 0) {
+                program = wtu.setupProgram(gl, ["vshader", "fshader_rgba_" + testcase.name], ["a_position", "a_coord"], [0, 1]);
+                format = gl.RGBA;
+            } else {
+                program = wtu.setupProgram(gl, ["vshader", "fshader_luminance_alpha"], ["a_position", "a_coord"], [0, 1]);
+                format = testcase.format;
+            }
+            gl.bindTexture(gl.TEXTURE_2D, texture[2]);
+            gl.texImage2D(gl.TEXTURE_2D, 0, gl.RGBA, width, height, 0, gl.RGBA, gl.UNSIGNED_BYTE, uint);
+            var fbo = gl.createFramebuffer();
+            gl.bindFramebuffer(gl.FRAMEBUFFER, fbo);
+            gl.framebufferTexture2D(gl.FRAMEBUFFER, gl.COLOR_ATTACHMENT0, gl.TEXTURE_2D, texture[2], 0);
+            if (gl.checkFramebufferStatus(gl.FRAMEBUFFER) == gl.FRAMEBUFFER_COMPLETE) {
+                gl.bindTexture(gl.TEXTURE_3D, texture[i]);
+                gl.texImage3D(gl.TEXTURE_3D, 0, format, width, height, depth, 0, format, gl.UNSIGNED_BYTE, uint2);
+                gl.copyTexSubImage3D(gl.TEXTURE_3D, 0, 0, 0, layer, 0, 0,width, height);
+
+                wtu.setupUnitQuad(gl, 0, 1);
+                var depthLoc = gl.getUniformLocation(program, "u_depth");
+                if(depthLoc) {
+                    gl.uniform1f(depthLoc, layer);
+                }
+                var sampler = gl.getUniformLocation(program, 'u_sampler0');
+                if(sampler) {
+                    gl.activeTexture(gl.TEXTURE0);
+                    setUpTexStatus();
+                    gl.uniform1i(sampler, 0);
+                    gl.clear(gl.COLOR_BUFFER_BIT);
+                }
+                wtu.drawUnitQuad(gl);
+
+                if(i == 0) {
+                    bytes = new Uint8Array(width * height * 4);
+                    gl.readPixels(0, 0, width, height, gl.RGBA, gl.UNSIGNED_BYTE, bytes);
```

Comment:
Why is it necessary to do a readPixels here? Isn't the intent that these bytes be equivalent to "uint"? Reading back this data from the framebuffer increases the likelihood that multiple bugs in the WebGL implementation may cause the test to pass.